radix-engine/tests/state_track.rs

```diff
@@ -0,0 +1,142 @@
+#[rustfmt::skip]
+pub mod test_runner;
+
+use crate::test_runner::TestRunner;
+use radix_engine::engine::CommitReceipt;
+use radix_engine::engine::RuntimeError;
+use radix_engine::ledger::PhysicalSubstateId;
+use scrypto::core::Network;
+use scrypto::prelude::*;
+use transaction::builder::ManifestBuilder;
+
+macro_rules! substate_id {
+    ($tx_hash:expr, $idx:expr) => {
+        PhysicalSubstateId(Hash::from_str($tx_hash).unwrap(), $idx)
+    };
+}
+
+#[test]
+fn test_state_track_success() {
+    // Arrange
+    let mut test_runner = TestRunner::new(true);
+    let (public_key, _, account) = test_runner.new_account();
+    let (_, _, other_account) = test_runner.new_account();
+
+    // Act
+    let manifest = ManifestBuilder::new(Network::LocalSimulator)
+        .withdraw_from_account(RADIX_TOKEN, account)
+        .call_method_with_all_resources(other_account, "deposit_batch")
+        .build();
+    let receipt = test_runner.execute_manifest(manifest, vec![public_key]);
+
+    // Assert
+    receipt.expect_success();
+    let mut expected_downs = HashSet::new();
+    expected_downs.extend([
+        substate_id!(
+            "5feceb66ffc86f38d952786c6d696c79c2dbc239dd4e91b46729d73a27fb57e9",
```

Comment:
The current substate_ids aren't very stable yet (most likely to change soon). Is there another less fragile way to test without knowing "expected_down" ids?

Reply:
Yes, this is ugly at the moment. I will try to enhance it once I have more fee logic implemented; then I can inspect the downed fee substates.

At the moment, it's more of a testing whether (not what) states have been rolled back/committed.